### ISSUE TITLE
Ability to have more then 1 Text object on a element.

### ISSUE THREAD

**Issue by luuxx** [QuestPDF/library#23]
**Is your feature request related to a problem? Please describe.**
We want to be able to change the style of single word, part of a sentence  or any where in-between without messing up the flow if the sentence.  

**Describe the solution you'd like**
The ability to stack Text objects so you can give different parts of sentence different stylings without interrupting the flow of the sentence.

**Describe alternatives you've considered**
Styling markup in a string. To allow for styling markdown like * to make a part bold or italic. Or a HTML to text convertor. 

**Additional context**
Basically we want full control over how the text in a row/column/element looks so the generated PDF to the vision we have for the final product.

**Comment by MarcinZiabek** [comment]
Hello :) Thank you for your suggestion. I have great news for you - the requested feature is already in progress, available in the `richtext` branch. I am hoping to release this feature in the October update. There are still a couple of corner cases to take care of.

Here you may find a short list of features that will be added:
- paging support,
- text background,
- text stroke,
- text underline,
- changing text style within a run (all style properties will be supported: weight, color, size, etc.),
- inserting links in text,
- inserting custom components in text, e.g. an image,
- inserting page numbers in text (current page, total pages, page of location),
- some text-rendering improvements (rendering text according to font settings). 

And an example Fluent API:
```csharp
.Text(text =>
{
    text.DefaultTextStyle(TextStyle.Default);
    text.AlignLeft();
    text.ParagraphSpacing(10);
    
    text.Span(Placeholders.LoremIpsum());
    
    text.NewLine();
    
    text.Span("This text is a normal text, ");
    text.Span("this is a bold text, ", TextStyle.Default.Bold());
    text.Span("this is a red and underlined text, ", TextStyle.Default.Color(Colors.Red.Medium).Underlined());
    text.Span("and this is slightly bigger text.", TextStyle.Default.Size(16));
    
    text.NewLine();
    
    text.Span("The new text element also supports injecting custom content between words: ");
    text.Element().PaddingBottom(-10).Height(16).Width(32).Image(Placeholders.Image);
    text.Span(".");
    
    text.NewLine();
    
    text.Span("This is page number ");
    text.CurrentPageNumber();
    text.Span(" out of ");
    text.TotalPages();
    
    text.NewLine();
    
    text.ExternalLocation("Please visit QuestPDF website", "https://www.questpdf.com");
    
    text.NewLine();
});
```

That produces the following result:

![image](https://user-images.githubusercontent.com/9263853/132867660-ad8c75d5-934e-4449-b04d-7f5205860ea8.png)

Please let me know what do you think :)

**Comment by aclimarin** [comment]
Great! I was looking for this.
@luuxx you have you done some kind of workaround on this?

**Comment by MarcinZiabek** [comment]
@aclimarin At this point, the best solution is to propose an alternative document layout that can be composed with current limitations. Or you can just skip the implementation of custom text styling for now and improve it in the nearest future.

**Can you please provide an example of what text styling you would like to achieve?**

**Comment by aclimarin** [comment]
I need to generate contracts and sometimes it has to have a bold text in the middle, like the properties of a client model. Like this:

**Comment by aclimarin** [comment]
I need to generate contracts and sometimes it has to have a bold text in the middle, like the properties of a client model. Like this:

This is a purchase contract of **Ana Caroline Lima**

**Comment by MarcinZiabek** [comment]
Thank you for your feedback. It makes sense, I miss that feature as well - text support is just very hard to implement hence taking a lot of time to make it right. As stated before, I am already working on it and I hope it will be available soon :)

**Comment by aclimarin** [comment]
How would work with an alternative layout?

**Comment by MarcinZiabek** [comment]
@luuxx @aclimarin I have just released a beta version of the QuestPDF 2021.10. It does contain text styling capabilities. Can you please test this version in your solutions and let me know if it works as expected? Any feedback will be highly appreciated :) 

https://www.nuget.org/packages/QuestPDF/2021.10.0-beta

**Comment by aclimarin** [comment]
@MarcinZiabek Thank you!

**Comment by MarcinZiabek** [comment]
This feature was implemented in the QuestPDF 2021.10 and is now publicly available in the stable channel :) Closing
